CMakeLists.txt

```diff
@@ -148,18 +153,35 @@ message(STATUS "Setting version to: ${MULTIPASS_VERSION}")
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
 set(CMAKE_CXX_EXTENSIONS OFF)
+if(MSVC)
+  add_definitions(-D_WIN32_WINNT=0xA00) # A gprc header requires this (specifies Windows 10)
+  add_definitions(-DNOMINMAX) # otherwise windows defines macros which interfere with std::min, std::max
+  add_definitions(-DMULTIPASS_PLATFORM_WINDOWS)
+  add_definitions(-DLIBSSH_STATIC) # otherwise adds declspec specifiers to libssh apis
+  add_definitions(-D_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS)
+  add_definitions(-DWIN32_LEAN_AND_MEAN)
+else()
+  add_compile_options(-Werror -Wall -pedantic -fPIC)
+
+  if(APPLE)
+    add_definitions(-DMULTIPASS_PLATFORM_APPLE)
+  else()
+    add_compile_options(-Wextra -Wempty-body -Wformat-security -Winit-self -Warray-bounds)
 
-add_compile_options(-Werror -Wall -pedantic -Wextra -Wempty-body -Wformat-security -Winit-self -Warray-bounds -fPIC)
+    if(NOT ${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "^arm")
+      add_compile_options(-Wcast-align)
+    endif()
 
-if(NOT ${CMAKE_HOST_SYSTEM_PROCESSOR} MATCHES "^arm")
-  add_compile_options(-Wcast-align)
-endif()
+    CHECK_CXX_COMPILER_FLAG("-Wno-expansion-to-defined" COMPILER_SUPPORTS_NO_EXP_TO_DEFINED)
+    if(COMPILER_SUPPORTS_NO_EXP_TO_DEFINED)
+      add_compile_options(-Wno-expansion-to-defined)
+    endif()
+
+    add_definitions(-DMULTIPASS_PLATFORM_LINUX)
+  endif()
 
-CHECK_CXX_COMPILER_FLAG("-Wno-expansion-to-defined" COMPILER_SUPPORTS_NO_EXP_TO_DEFINED)
-if(COMPILER_SUPPORTS_NO_EXP_TO_DEFINED)
-  add_compile_options(-Wno-expansion-to-defined)
+  add_compile_options(-Wno-error=deprecated-declarations)
```

Comment:
Could this, and the `add_` further up, be combined, or do they need to come in order?

Reply:
I wasn't sure. I see this in the man:

       You can mix options and other arguments.  For the most part, the order you use doesn't
       matter.  Order does matter when you use several options of the same kind; for example, if
       you specify -L more than once, the directories are searched in the order specified.  Also,
       the placement of the -l option is significant.

Let me have a better look.

Reply:
You're right, `-Wno-error=deprecated-declarations` can even appear before `-Werror`. Fix coming.

Reply:
Done, amended the last commit.